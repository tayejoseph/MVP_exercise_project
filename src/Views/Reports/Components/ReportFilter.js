import React from 'react'
import { InputGroup } from 'UI'

const ReportFilter = (props) => {
  const {
    filterData,
    handleInput,
    projectLists,
    gateWayLists,
    handleGenerateReport,
  } = props

  return (
    <>
      <InputGroup
        type="select"
        name="projectId"
        value={filterData.projectId}
        onChange={handleInput}
        optionLists={
          <>
            <option value="" disabled>
              Select project
            </option>
            <option value={'all'}>All Project</option>
            {projectLists
              ? projectLists.map((item) => (
                  <option value={item.projectId} key={item.projectId}>
                    {item.name}
                  </option>
                ))
              : null}
          </>
        }
      />
      <InputGroup
        type="select"
        name="gatewayId"
        className="gateway-item"
        value={filterData.gatewayId}
        optionLists={
          <>
            <option value="" disabled>
              Select gateway
            </option>
            <option value={'all'}>All Gateway</option>
            {gateWayLists
              ? gateWayLists.map((item) => (
                  <option value={item.gatewayId} key={item.gatewayId}>
                    {item.name}
                  </option>
                ))
              : null}
          </>
        }
        onChange={handleInput}
      />
      <InputGroup
        type="date"
        flexLabel="From"
        value={filterData.from}
        onChange={handleInput}
        min="2021-01-01"
        max="2021/12/31"
        name="from"
      />
      <InputGroup
        type="date"
        flexLabel="To"
        value={filterData.to}
        onChange={handleInput}
        min="2021-01-01"
        max="2021/12/31"
        name="to"
      />
      <button onClick={() => handleGenerateReport(filterData)}>
        Generate Report
      </button>
    </>
  )
}

export default ReportFilter
