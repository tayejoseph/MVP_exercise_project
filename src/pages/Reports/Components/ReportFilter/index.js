import React from 'react';
import PropTypes from 'prop-types';
import { InputGroup } from 'components';

const ReportFilter = (props) => {
  const { filterData, handleInput, projectLists, gateWayLists, handleGenerateReport } = props;

  return (
    <>
      <InputGroup
        type="select"
        name="projectId"
        value={filterData.projectId}
        onChange={handleInput}
        optionLists={
          <>
            <option value="" disabled>
              Select project
            </option>
            <option value={'all'}>All Project</option>
            {projectLists.length > 0
              ? projectLists.map((item) => (
                  <option value={item.projectId} key={item.projectId}>
                    {item.name}
                  </option>
                ))
              : null}
          </>
        }
      />
      <InputGroup
        type="select"
        name="gatewayId"
        className="gateway-item"
        value={filterData.gatewayId}
        optionLists={
          <>
            <option value="" disabled>
              Select gateway
            </option>
            <option value={'all'}>All Gateway</option>
            {gateWayLists.length > 0
              ? gateWayLists.map((item) => (
                  <option value={item.gatewayId} key={item.gatewayId}>
                    {item.name}
                  </option>
                ))
              : null}
          </>
        }
        onChange={handleInput}
      />
      <InputGroup
        type="date"
        flexLabel="From"
        value={filterData.from}
        onChange={handleInput}
        // min="1900-01-01" max="2016-06-06"
        min="2021-01-01"
        max="2021-12-31"
        name="from"
      />
      <InputGroup
        type="date"
        flexLabel="To"
        value={filterData.to}
        onChange={handleInput}
        min="2021-01-01"
        max="2021-12-31"
        name="to"
      />
      <button onClick={() => handleGenerateReport(filterData)}>Generate Report</button>
    </>
  );
};

ReportFilter.propTypes = {
  filterData: PropTypes.shape({
    gatewayId: PropTypes.string,
    projectId: PropTypes.string,
    from: PropTypes.string,
    to: PropTypes.string
  }),
  handleInput: PropTypes.func,
  projectLists: PropTypes.array,
  gateWayLists: PropTypes.array,
  handleGenerateReport: PropTypes.func
};

export default ReportFilter;
