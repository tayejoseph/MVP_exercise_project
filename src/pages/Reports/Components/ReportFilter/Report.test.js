import renderer from 'react-test-renderer';
import ReportFilter from './index';

it('renders correctly', () => {
  const filterData = {
    gatewayId: '',
    projectId: '',
    from: '2021-01-01',
    to: '2021-12-31'
  };
  const tree = renderer
    .create(
      <ReportFilter
        {...{
          filterData,
          handleInput: () => ({}),
          projectLists: [],
          gateWayLists: [],
          handleGenerateReport: () => ({})
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
