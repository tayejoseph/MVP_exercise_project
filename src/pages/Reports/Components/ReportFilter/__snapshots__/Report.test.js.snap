// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="sc-bcXHqe jzXHmT input-group "
    id=""
  >
    <div
      className="input-container"
    >
      <select
        autoComplete="off"
        disabled={false}
        name="projectId"
        onChange={[Function]}
        required={false}
        value=""
      >
        <option
          disabled={true}
          value=""
        >
          Select project
        </option>
        <option
          value="all"
        >
          All Project
        </option>
      </select>
    </div>
  </div>,
  <div
    className="sc-bcXHqe jzXHmT input-group gateway-item"
    id=""
  >
    <div
      className="input-container"
    >
      <select
        autoComplete="off"
        disabled={false}
        name="gatewayId"
        onChange={[Function]}
        required={false}
        value=""
      >
        <option
          disabled={true}
          value=""
        >
          Select gateway
        </option>
        <option
          value="all"
        >
          All Gateway
        </option>
      </select>
    </div>
  </div>,
  <div
    className="sc-bcXHqe jzXHmT input-group "
    id=""
  >
    <div
      className="input-container"
    >
      <span
        className="flex-label"
      >
        From
      </span>
      <input
        autoComplete="off"
        disabled={false}
        name="from"
        onChange={[Function]}
        placeholder=""
        required={false}
        type="date"
        value="2021-01-01"
      />
    </div>
  </div>,
  <div
    className="sc-bcXHqe jzXHmT input-group "
    id=""
  >
    <div
      className="input-container"
    >
      <span
        className="flex-label"
      >
        To
      </span>
      <input
        autoComplete="off"
        disabled={false}
        name="to"
        onChange={[Function]}
        placeholder=""
        required={false}
        type="date"
        value="2021-12-31"
      />
    </div>
  </div>,
  <button
    onClick={[Function]}
  >
    Generate Report
  </button>,
]
`;
